include/gunrock/algorithms/async/async_enactor.hxx

```diff
@@ -0,0 +1,68 @@
+
+#pragma once
+
+#include <gunrock/algorithms/async/queue.cuh>
+
+namespace gunrock {
+  
+template <typename algorithm_problem_t>
+struct async_enactor_t {
+
+  using vertex_t = typename algorithm_problem_t::vertex_t;
+  using edge_t   = typename algorithm_problem_t::edge_t;
+  using weight_t = typename algorithm_problem_t::weight_t;
+  
+  using single_queue_t = uint32_t;
+  using queue_t        = MaxCountQueue::Queues<vertex_t, single_queue_t>;
+  queue_t q;
+  
+  algorithm_problem_t* problem;
+  std::shared_ptr<cuda::multi_context_t> context;
+
+  async_enactor_t(const async_enactor_t& rhs) = delete;
+  async_enactor_t& operator=(const async_enactor_t& rhs) = delete;
+
+  // !! These parameters may be application specific, so really we want a way to 
+  // pass them in as "execution policy parameters"
+  // For instance ... I don't think these parameters would work w/ async pagerank
+  int num_queue       = 4;
+  int min_iter        = 800;
+  int num_block       = 56 * 5;
+  int num_thread      = 256;
+  float sizing_factor = 1.5;
+
```

Comment:
The user needs control over these parameters -- these are basically hardcoded for BFS at the moment.  They'd break for PageRank.  What's the best API to give control of these kinds of "execution policy parameters" to the user?  I think there's an analog in synchronous essentials, but I've never seen it used.